src/main/java/com/alphawallet/attestation/NFTAttestation.java

```diff
@@ -29,7 +40,15 @@ public NFTAttestation(SignedAttestation att, ERC721Token nftToken) {
         }
     }
 
+    public byte[] getPreHash() {
+        return SignatureUtility.convertToPersonalEthMessage(getDerEncoding());
```

Comment:
This is a bit inconsistent with what is done in Attestation and the other classes. That is, in Attestation the raw DER data is returned and then it is the responsibility of the signing method to add the Personal prefix if needed. This is for example done in AttestedObject. But maybe the prehash in Attestation should return the DER data with the Personal prefix added?

Reply:
Yes that's a good point. It's going to be a bit awkward at some end whichever way we do it; if you add the signpersonal prefix in the attestation, when you go to check it at the other end, a dapp developer might be concerned why they aren't checking via signpersonal, however the request was made via signpersonal. 

It was a quandary whether to do this or not - I thought it would make more sense for the attestation to appear to the dapp developer as being needed to be wrapped, since they're going to have to call signpersonalmessage instead of just 'sign'.